### PR TITLE
skip routing invalid segments

### PR DESCRIPTION
**Pull request by kianzarrin**
I moved the condition for checking invalid section to the top level. I don't know why we calculate `incomingVehicleLanes`++ and `outgoingVehicleLanes++` when segment is invalid. I think we should skip it.
This makes the code simpler and easier to apply my future changes.

Hide white spaces when reviewing.

### CODE REVIEW

**Comment by kianzarrin** [comment]
need to do test

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 1]
```diff
@@ -933,8 +921,7 @@ public class RoutingManager
                                         (nextIncomingDir == ArrowDirection.Right && hasLeftArrow) || // valid incoming right
                                         (nextIncomingDir == ArrowDirection.Left && hasRightArrow) || // valid incoming left
                                         (nextIncomingDir == ArrowDirection.Forward && hasForwardArrow) || // valid incoming straight
-                                        (nextIncomingDir == ArrowDirection.Turn && canTurn) /*valid turning lane*/) 
-                                    {
+                                        (nextIncomingDir == ArrowDirection.Turn && canTurn) /*valid turning lane*/) {
```
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:924]: auto formatting has done this when I added the final bracket. ignore if there is only white space changes.

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 2]
```diff
@@ -998,8 +985,7 @@ public class RoutingManager
 
                                     // routed vehicle that does not follow lane arrows (trains, trams,
                                     // metros, monorails)
-                                    // TODO [issue #1053] this causes cars to turn on mixed car/track lanes against lane arrows
```
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:1001]: this is not true

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 3]
```diff
@@ -2107,11 +2082,20 @@ bool laneChangesAllowed
...
-                    totalIncomingLanes += incomingVehicleLanes;
-                    totalOutgoingLanes += outgoingVehicleLanes;
+                    if (nextSegmentId != prevSegmentId) {
+                        totalIncomingLanes += incomingVehicleLanes;
+                        totalOutgoingLanes += outgoingVehicleLanes;
+                    }
+                } else {
+                    // invalid segment
+                    if (extendedLogRouting) {
+                        if (extendedLogRouting) {
```
Comment (aubergine10) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:2093]: wat

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:2093]: must have some merging error. Ill fix that.

**Comment by kianzarrin** [comment]
I performed several tests involving building/deleting/spliting/reversing segment at junction and did not manage to break lane routing.

**Review by aubergine10 — APPROVED** [review]
LGTM 👍 

Only did basic testing in-game but didn't notice any problems.

**Review by krzychu124 — APPROVED** [review]
Looks good, I didn't notice any issues in game, did a quick smoke tests with an overlay tool for lane transitions. Why this PR has **29 commits**... I have no idea.

**Comment by kianzarrin** [comment]
> Why this PR has 29 commits... I have no idea.

merging! its been 2~3 weeks

**Pull request merged**